### PR TITLE
Add models to wrap the JSON payload and conversion of a list item to a db entry

### PR DESCRIPTION
**Pull request by amaalali**
- Add model to encapulate interacting with the Json payload.
- Add config for Mongo with integration tests
- Add BaseGenerators helper trait
- Add conversion to a model that will add the list name and the list meta-data to each list entry
- Add JSON serialization of GenericListItem

### CODE REVIEW

File: app/models/GenericListItem.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (chris-ianson) [inline app/models/GenericListItem.scala:1]: Is this ReferenceDataList renamed/updated?

File: app/models/MongoDateTimeFormats.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (chris-ianson) [inline app/models/MongoDateTimeFormats.scala:1]: Do we need a spec for this?

File: test/models/ReferenceDataPayloadSpec.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+import base.SpecBase
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.libs.json.JsObject
+import play.api.libs.json.Json
+
+class ReferenceDataPayloadSpec extends SpecBase with ScalaCheckDrivenPropertyChecks {
+
+  "listsNames" - {
+    "should return an empty Seq when there lists" in {
```
Comment (chris-ianson) [inline test/models/ReferenceDataPayloadSpec.scala:29]: should this be _when there are no lists_

File: test/models/ReferenceDataPayloadSpec.scala [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+      val referenceDataPayload = ReferenceDataPayload(data)
+      val listNames            = referenceDataPayload.listsNames
+
+      listNames mustEqual Set(referenceDataPayload.ListName("testListName1"), referenceDataPayload.ListName("testListName2"))
+    }
+  }
+
+  "getlist" - {
+    "returns the list for a listName" in {
```
Comment (chris-ianson) [inline test/models/ReferenceDataPayloadSpec.scala:81]: Do we need a test for when listEntries is empty?

**Review by chris-ianson — CHANGES_REQUESTED** [review]
Couple of minor comments questions

File: app/models/GenericListItem.scala (hunk 1 above)
Comment (amaalali) [reply app/models/GenericListItem.scala:1]: No, there is still `ReferenceDataList`, this is just an intermediate object that enriches each items with the extra data needed for indexing and querying, since we're going with a single collection for items

File: app/models/MongoDateTimeFormats.scala (hunk 2 above)
Comment (amaalali) [reply app/models/MongoDateTimeFormats.scala:1]: Oops forgot to copy that over too

File: test/models/ReferenceDataPayloadSpec.scala (hunk 4 above)
Comment (amaalali) [reply test/models/ReferenceDataPayloadSpec.scala:81]: I'm not sure, I'm trying something new out where ListName is returned by ReferenceDataPayload and it's type is a path dependent type ... so I'm not sure if that's enough to ensure that we will return a value here

Comment (chris-ianson) [reply test/models/ReferenceDataPayloadSpec.scala:81]: 👍

**Review by chris-ianson — APPROVED** [review]

**Comment by chris-ianson** [comment]
Approved but you still need to update test and add datetime spec
